User meeting routes

inathan44 opened the pull request.

pbryan9 commented on server/api/routes/userMeeting.cjs:
```diff
@@ -1,33 +1,107 @@
+      const meeting = await Meeting.findByPk(req.params.meetingId);
+      if (!meeting) {
+        return res.status(404).send('No meeting found.');
+      } else if (req.user.id !== meeting.userId && req.user.role !== 'admin') {
```
req.user.id can be either meeting.userId OR meeting.buddyId

pbryan9 commented on server/api/routes/userMeeting.cjs:
```diff
@@ -1,33 +1,107 @@
-
-  res.send(`hello from /api/user/${req.params.userId}/meeting`);
+  try {
+    const userMeetings = await Meeting.findAll({
```
(just taking the note from your own catch - well done) -- need to check userId for existence (User.findByPk(:userId))

pbryan9 commented on server/api/routes/userMeeting.cjs:
```diff
@@ -1,33 +1,107 @@
+      const meeting = await Meeting.findByPk(req.params.meetingId);
+      if (!meeting) {
+        return res.status(404).send('No meeting found to update.');
+      } else if (req.user.id !== meeting.userId && req.user.role !== 'admin') {
```
need buddy access here as well

pbryan9 commented on server/api/routes/userMeeting.cjs:
```diff
@@ -1,33 +1,107 @@
+      }
+    } catch (error) {
+      console.error(
+        'Error trying to change meeting date. Please try again later.'
```
close*

inathan44 commented:
Added a check to the routes that have userId and meetingId in the route to make sure that the meeting that was being queried belonged to that user. Before this check, an admin could change a meeting (for ex meeting 15) even though they passed in a user is that DID NOT belong to that meeting.

A new middleware is added in to check if the user in params is a valid user.

queries now pull meetings where user is a buddy and allows the buddy to update meetings as well.

console.error bug was also fixed

inathan44 commented on server/api/routes/userMeeting.cjs:
```diff
@@ -1,33 +1,161 @@
+          .send('No meeting found matching user and meetingId to update.');
+      } else if (
+        req.user.id !== meeting.userId &&
+        req.user.id !== meeting.buddyId &&
```
Remove buddy ability to change the time of the meeting
